tests/testthat/test-gpu.R

```diff
@@ -0,0 +1,60 @@
+context("gpu")
+
+test_that("can generate GPU interface and pass arguments", {
+  skip_if_not_installed("mockery")
+  mock_odin_dust <- mockery::mock(carehomes, cycle = TRUE)
+
+  mockery::stub(compile_gpu, "odin.dust::odin_dust_", mock_odin_dust)
+
+  expect_identical(
+    compile_gpu(),
+    carehomes)
+  mockery::expect_called(mock_odin_dust, 1L)
+  expect_equal(
+    mockery::mock_args(mock_odin_dust)[[1]],
+    list(sircovid_file("odin/carehomes.R"), real_t = "float", gpu = TRUE))
+
+  compile_gpu(real_t = "double", rewrite_dims = TRUE,
+              gpu = FALSE, gpu_generate = TRUE)
```

Comment:
Does it make sense to try some of the other CUDA options here? e.g. fast_math, profile

Reply:
I don't think so - this sort of testing should be done in dust (where we don't suffer 1 minute compile times!), and is of relatively limited value unless we check that they had an effect. We do an interaction test already which confirms that the options would be passed